### PR TITLE
structured suggestion for E0223 ambiguous associated type

### PR DESCRIPTION
**Pull request by zackmdavis**
(routine (and when are we going to be done finding these, anyway?) but something that stuck out to me while glancing at #54970)

r? @estebank

### CODE REVIEW

**Comment by zackmdavis** [comment]
Wait, @estebank is currently the assignee for four other open PRs; we can do reviewer load-balancing better than this

r? @davidtwco

File: src/librustc_typeck/astconv.rs [hunk 1]
```diff
@@ -1093,10 +1093,12 @@ impl<'o, 'gcx: 'tcx, 'tcx> dyn AstConv<'gcx, 'tcx>+'o {
                                         name: &str) {
         struct_span_err!(self.tcx().sess, span, E0223, "ambiguous associated type")
             .span_label(span, "ambiguous associated type")
```
[the remark this answers is not in the document]
Comment (davidtwco) [reply src/librustc_typeck/astconv.rs]: I agree, with the change to a suggestion, this no longer improves the error. 

r=me when this is changed.

File: src/test/ui/error-codes/E0223.stderr [hunk 2]
```diff
@@ -2,9 +2,10 @@ error[E0223]: ambiguous associated type
   --> $DIR/E0223.rs:14:14
    |
 LL |     let foo: MyTrait::X;
-   |              ^^^^^^^^^^ ambiguous associated type
-   |
-   = note: specify the type using the syntax `<Type as MyTrait>::X`
+   |              ^^^^^^^^^^
+   |              |
+   |              ambiguous associated type
+   |              help: use fully-qualified syntax: `<Type as MyTrait>::X`
```
Comment (estebank) [inline src/test/ui/error-codes/E0223.stderr]: Is `Type` a default? If it is, then it shouldn't be machine applicable in those cases.

File: src/test/ui/did_you_mean/bad-assoc-ty.stderr [hunk 3]
```diff
@@ -82,25 +86,28 @@ error[E0223]: ambiguous associated type
...
   --> $DIR/bad-assoc-ty.rs:37:10
    |
 LL | type G = 'static + (Send)::AssocTy;
-   |          ^^^^^^^^^^^^^^^^^^^^^^^^^ ambiguous associated type
-   |
-   = note: specify the type using the syntax `<(dyn std::marker::Send + 'static) as Trait>::AssocTy`
+   |          ^^^^^^^^^^^^^^^^^^^^^^^^^
+   |          |
+   |          ambiguous associated type
+   |          help: use fully-qualified syntax: `<(dyn std::marker::Send + 'static) as Trait>::AssocTy`
```
Comment (estebank) [inline src/test/ui/did_you_mean/bad-assoc-ty.stderr]: Can you verify that this suggestion is actually accepted code?

File: src/test/ui/error-codes/E0223.stderr (hunk 2 above)
Comment (zackmdavis) [reply src/test/ui/error-codes/E0223.stderr]: [Good](https://github.com/rust-lang/rust/blob/77af314083e5acabf9ba5335e47271f35eef2e99/src/librustc_typeck/astconv.rs#L1288) [catch](https://github.com/rust-lang/rust/blob/77af314083e5acabf9ba5335e47271f35eef2e99/src/librustc_typeck/astconv.rs#L1242) :baseball: :broken_heart:

**Pull request merged**